lib/src/fnc/string.rs

```diff
@@ -81,5 +84,5 @@ pub fn uppercase(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
 }
 
 pub fn words(_: &Context, mut args: Vec<Value>) -> Result<Value, Error> {
-	Ok(args.remove(0).as_string().split(' ').collect::<Vec<&str>>().into())
+	Ok(args.remove(0).as_string().split_whitespace().collect::<Vec<&str>>().into())
```

Comment:
This is the bugfix mentioned in the title of the PR. Using `split(' ')` has unintended consequences when there are extra spaces before, in the middle, or after
```sql
> select * from string::words("  foo   bar  ");
[{"time":"85.534µs","status":"OK","result":["","","foo","","","bar","",""]}]
```

Reply:
@finnbear could we pull this out into its own PR? Just because it's a bugfix and not optimisation/cleanup/refactor...

Reply:
> @finnbear could we pull this out into its own PR? Just because it's a bugfix and not optimisation/cleanup/refactor...

Good idea, see https://github.com/surrealdb/surrealdb/pull/72 :tada: